server/src/test/java/org/elasticsearch/action/bulk/TransportShardBulkActionTests.java

```diff
@@ -509,329 +589,296 @@ public void testUpdateReplicaRequestWithConflictFailure() throws Exception {
-        DocWriteResponse.Result deleteWriteResult = DocWriteResponse.Result.DELETED;
-        IndexRequest indexRequest = new IndexRequest("index", "_doc", "id");
-        DeleteRequest deleteRequest = new DeleteRequest("index", "_doc", "id");
-        UpdateHelper.Result translate = new UpdateHelper.Result(indexRequest, docWriteResult,
-                new HashMap<String, Object>(), XContentType.JSON);
-        UpdateHelper.Result translateDelete = new UpdateHelper.Result(deleteRequest, deleteWriteResult,
-                new HashMap<String, Object>(), XContentType.JSON);
+        assertTrue(shard.isSyncNeeded());
 
-        BulkItemRequest[] itemRequests = new BulkItemRequest[1];
-        itemRequests[0] = request;
-
-        BulkItemResultHolder holder = TransportShardBulkAction.processUpdateResponse(updateRequest,
-                "index", indexResult, translate, shard, 7);
-
-        assertTrue(holder.isVersionConflict());
-        assertThat(holder.response, instanceOf(UpdateResponse.class));
-        UpdateResponse updateResp = (UpdateResponse) holder.response;
-        assertThat(updateResp.getGetResult(), equalTo(null));
-        assertThat(holder.operationResult, equalTo(indexResult));
-        BulkItemRequest replicaBulkRequest = holder.replicaRequest;
-        assertThat(replicaBulkRequest.id(), equalTo(7));
-        DocWriteRequest<?> replicaRequest = replicaBulkRequest.request();
-        assertThat(replicaRequest, instanceOf(IndexRequest.class));
-        assertThat(replicaRequest, equalTo(indexRequest));
-
-        BulkItemResultHolder deleteHolder = TransportShardBulkAction.processUpdateResponse(updateRequest,
-                "index", deleteResult, translateDelete, shard, 8);
-
-        assertFalse(deleteHolder.isVersionConflict());
-        assertThat(deleteHolder.response, instanceOf(UpdateResponse.class));
-        UpdateResponse delUpdateResp = (UpdateResponse) deleteHolder.response;
-        assertThat(delUpdateResp.getGetResult(), equalTo(null));
-        assertThat(deleteHolder.operationResult, equalTo(deleteResult));
-        BulkItemRequest delReplicaBulkRequest = deleteHolder.replicaRequest;
-        assertThat(delReplicaBulkRequest.id(), equalTo(8));
-        DocWriteRequest<?> delReplicaRequest = delReplicaBulkRequest.request();
-        assertThat(delReplicaRequest, instanceOf(DeleteRequest.class));
-        assertThat(delReplicaRequest, equalTo(deleteRequest));
+        // if we sync the location, nothing else is unsynced
+        CountDownLatch latch = new CountDownLatch(1);
+        shard.sync(context.getLocationToSync(), e -> {
+            if (e != null) {
+                throw new AssertionError(e);
+            }
+            latch.countDown();
+        });
+
+        latch.await();
+        assertFalse(shard.isSyncNeeded());
 
         closeShards(shard);
     }
+    //    public void testCalculateTranslogLocation() throws Exception {
```

Comment:
what about this?

Reply:
Forgot to remove the commented out version. It now lives as BulkPrimaryExecutionContextTests#testTranslogLocation